### DISCUSSION TITLE
How to update label on an animated plot without it overlaying

### DISCUSSION THREAD

**Discussion by wiseac** [JuliaPlots/Makie.jl#2064] [Q&A] [answered]
Hello, 
I'm currently trying to add a label that updates with an animated plot/ plot that is always updating via a circular buffer. The overlay looks something like this picture. I want the newest temperature reading to be updated in the label (which it does but just overlays).

![GlMakieOverlay](https://user-images.githubusercontent.com/89418908/174217948-790f3bdc-8983-4676-a1c6-f0ff1e671fde.png)

My main code is in a "while" loop for pushing the updated temperatures but my label code looks like this..
`Label(fig[1, 1], "Temperature (°C) " * string(t[][500]), 
    halign=:right, textsize=18, padding=(5, 5, 5, 5), tellheight= false, 
    tellwidth = false, valign = :top)`

t[][500] is just defining the latest value in my circular buffer.

I saw the @lift function in the animated documentation but im not sure how to use it or if it is even applicable.

Thanks,
W

**Comment by asinghvi17** [comment, marked as answer]
You can directly set the label's text in your animation, assuming `lab::Label`:
```julia
lab.text[] = "Temperature (°C) " * string(t[][500])
```
this should "just work".

**Reply by wiseac** [reply]
Hello,
So my code now looks like,
`   lab = Label(fig[1, 1],
    halign=:right, textsize=18, padding=(5, 5, 5, 5), tellheight= false, 
    tellwidth = false, valign = :top)
    lab.text[] = "Temperature (°C) " * string(t[][500])`

but it still causes that overlay, should i assign the label outside of the "while" loop?

**Reply by wiseac** [reply]
I just did that and it Worked!

Thank you very much!

**Comment by BeastyBlacksmith** [comment]
Somehow this does not work for me, if the Label starts with an empty string like:
```julia
julia> f = Figure()
julia> lab = Label(
               f[1, 1], ""
           )
julia> lab.text[] = "WOW!"
```

**Reply by asinghvi17** [reply]
Can confirm, this should probably be an issue
